### ISSUE TITLE
Support for RoomVisual

### ISSUE THREAD

**Issue by thmsndk** [thmsndk/Screeps3D#30]
this needs to be fletched out in regards to how it should work.

rendering "text" above creeps and such makes sense, what about rectangles and other stuff.
should it render as "sprites" is it a "hud"? like what OverMind does? how does it should it render relative to the position? is it a  map overiview overlay?" all theese questions, needs to be refined :)

https://docs.screeps.com/api/#RoomVisual

**Comment by Robalian** [comment]
You can't say for sure which visuals are used for overlay and which ones are related to position in room. For example overmind uses rectangles, text, maybe lines for its overlay. But you can use the same things for visualizing which resources are in which lab (https://github.com/screepers/RoomVisual), visualize creep paths etc.

I guess visuals should be mapped to room plane, rendered slightly above everything else.

**Comment by thmsndk** [comment]
@Robalian Yeah, that was my last idea, to have some sort of "mapoverlay" or perhaps as you describe, an overlay  between room level and map level.

Not sure if we can determine some of the visuals if they are mapped to a specific object, e.g creeps, and if they are, just render it above that object.

**Comment by thmsndk** [comment]
duckymirror suggested something like this, but basicly having roomvisuals as a "holograph" where viewing from top-down it's more solid, and then it renders all the way to the ground in an alpha blended manner.

It should also be controlled by an option so you can toggle the view/overlay on and off.

![image](https://user-images.githubusercontent.com/9084377/74775488-1fc81600-5296-11ea-8aa2-feac5bafca34.png)
